web/packages/shared/components/MenuLoginWithActionMenu/MenuLoginWithActionMenu.tsx

```diff
@@ -0,0 +1,130 @@
+  LoginItem,
+  MenuInputType,
+  MenuLogin,
+} from 'shared/components/MenuLogin';
+
+/**
+ * Displays a menu button with additional action menu attached to the right of it.
+ * <MenuLoginWithActionMenu> expects two children, a LoginItem for <MenuLogin> and <MenuItem>
+ * for action memnu.
+ *
+ * @example
+ * <MenuLoginWithActionMenu
+ *   buttonText="Log In"
+ *   size="medium"
+ *   onSelect={() => alert('item selected')}
+ *   getLoginItems= {() => LoginItem[]}
+ * >
+ *   <MenuItem>Foo</MenuItem>
+ *   <MenuItem>Bar</MenuItem>
+ * </MenuLoginWithActionMenu>
+ */
+export const MenuLoginWithActionMenu = ({
+  buttonText,
+  onSelect,
+  getLoginItems,
+  children,
+  width,
+  size = 'medium',
+  placeholder,
+  inputType,
+}: {
+  /** button text for main menu button. */
+  buttonText: string;
+  /** handles select or click in main menu items. */
+  onSelect: (e: React.SyntheticEvent, login: string) => void;
+  /** fetches login items. */
+  getLoginItems: () => LoginItem[] | Promise<LoginItem[]>;
+  /** action menu items. */
+  children: MenuItemComponent | MenuItemComponent[];
+  width?: string;
+  size?: ButtonSize;
+  /** text for action menu search box or static label.  */
+  placeholder?: string;
+  /** input type for menu item filter input. */
+  inputType?: MenuInputType;
+}) => {
+  const moreButtonRef = useRef<HTMLButtonElement>(null);
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <Flex>
+      <MenuLogin
+        width={width}
+        inputType={inputType}
+        onSelect={onSelect}
+        textTransform="none"
+        alignButtonWidthToMenu
+        getLoginItems={getLoginItems}
+        transformOrigin={{
+          vertical: 'top',
+          horizontal: 'right',
+        }}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'right',
+        }}
+        style={{ borderTopRightRadius: 0, borderBottomRightRadius: 0 }}
+        buttonText={buttonText}
+        placeholder={placeholder}
+      />
+      <ButtonBorder
+        setRef={moreButtonRef}
+        px={1}
+        size={size}
+        onClick={() => setIsOpen(true)}
+        /* css value matches with ButtonWithMenu component */
+        css={`
+          border-left: none;
+          border-top-left-radius: 0;
+          border-bottom-left-radius: 0;
+        `}
+        title="Open menu"
+      >
+        <icons.MoreVert size={size} color="text.slightlyMuted" />
+      </ButtonBorder>
+      <Menu
+        anchorEl={moreButtonRef.current}
+        open={isOpen}
+        onClose={() => setIsOpen(false)}
+        getContentAnchorEl={null}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'right',
+        }}
+        transformOrigin={{
+          vertical: 'top',
+          horizontal: 'right',
+        }}
+        menuListProps={{ onClick: () => setIsOpen(false) }}
+      >
+        {children}
+      </Menu>
+    </Flex>
+  );
+};
+
+type MenuItemComponent = ReactElement<typeof MenuItem>;
```

Comment:
Let's copy the same caveat comment from `ButtonWithMenu` here. I read it again now and I'm not sure if it's still true or not. 😂 But we can inspect that some other day.